### PR TITLE
[move-to-yul] Setup experiments.

### PR DESCRIPTION
**Pull request by wrwg**
This PR sets up experiments for code generation and wires them to the test infra. This will be needed for the upcoming control flow graph optimizations, and down the road also for other things.

A new module `experiments.rs` is added which declares constants for experiments. An experiment can be activated on the command line with `--experiment=<name>`. In a baseline tests, one can add the comment
```
// experiment: <name>
```

... to the source. The test is then run both with and without the experiment, producing different baseline files.

A first experiment is used to suppress generation of source info in baseline tests by default. This is useful because a lot of useless diffs are created if source positions only slightly shift. Source info will now only be generated during tests if `--experiment=capture-source-info` is set. Only one of our tests(`Resources.move`)  has this on, in order to ensure source info keeps being tested.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/diem/diem/blob/main/CONTRIBUTING.md#pull-requests)?

Yes

### CODE REVIEW

**Review by rahxephon89 — APPROVED** [review]

File: language/evm/move-to-yul/tests/Resources.exp.capture-source-info [hunk 1]
```diff
@@ -0,0 +1,1269 @@
+/* =======================================
```
[the remark this answers is not in the document]
Comment (wrwg) [reply language/evm/move-to-yul/tests/Resources.exp.capture-source-info:1]: Just tried the same on a freshly synced client and the file was re-generated.

[the remark this answers is not in the document]
Comment (gegrid) [reply language/evm/move-to-yul/tests/Resources.exp.capture-source-info:1]: Interesting, I noticed that some tests also failed

```
failures:

---- test_dispatch_revert stdout ----
Error: failed to find path to solc -- is the environment variable SOLC_EXE set?
thread 'test_dispatch_revert' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/test/src/lib.rs:194:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- test_dispatch_basic stdout ----
Error: failed to find path to solc -- is the environment variable SOLC_EXE set?
thread 'test_dispatch_basic' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/test/src/lib.rs:194:5

---- test_dispatch_fallback stdout ----
Error: failed to find path to solc -- is the environment variable SOLC_EXE set?
thread 'test_dispatch_fallback' panicked at 'assertion failed: `(left == right)`
  left: `1`,
 right: `0`: the test returned a termination value with a non-zero status code (1) which indicates a failure', /rustc/59eed8a2aac0230a8b53e89d4e99d55912ba6b35/library/test/src/lib.rs:194:5
```

I am on a clean main branch, did I miss some local setups? Thanks for the quick reply!

[the remark this answers is not in the document]
Comment (gegrid) [reply language/evm/move-to-yul/tests/Resources.exp.capture-source-info:1]: we can chat over our 1:1 as well :)

[the remark this answers is not in the document]
Comment (wrwg) [reply language/evm/move-to-yul/tests/Resources.exp.capture-source-info:1]: You need to install the solc compiler:

```
./scripts/dev_setup.sh -dp
source ~/.profile
```

**Pull request merged**